src/bin/pg_autoctl/cli_config.c

```diff
@@ -130,9 +175,54 @@ cli_config_check(int argc, char **argv)
 
 			(void) cli_config_check_pgsetup(&(config.pgSetup));
 
-			fprintf(stdout, "Current Configuration (includes pgSetup):\n");
-			keeper_config_write(stdout, &config);
-			fprintf(stdout, "\n");
+			if (outputJSON)
+			{
+				JSON_Value *js = json_value_init_object();
+				JSON_Value *jsPostgres = json_value_init_object();
+				JSON_Value *jsKConfig = json_value_init_object();
+
+				JSON_Object *root = json_value_get_object(js);
+				JSON_Object *jsPostgresObj = json_value_get_object(jsPostgres);
+				JSON_Object *jsKConfigObj = json_value_get_object(jsKConfig);
+
+				char *serialized_string;
+				int len;
+
+				/* prepare both JSON objects */
+				if (!pg_setup_as_json(&(config.pgSetup), jsPostgresObj))
+				{
+					/* can't happen */
+					exit(EXIT_CODE_INTERNAL_ERROR);
+				}
+
+				if (!keeper_config_to_json(jsKConfigObj, &config))
+				{
+					log_fatal("Failed to serialize monitor configuration to JSON");
+					exit(EXIT_CODE_BAD_CONFIG);
+				}
+
+				/* concatenate JSON objects into a container object */
+				json_object_set_value(root, "postgres", jsPostgres);
+				json_object_set_value(root, "config", jsKConfig);
+
+				/* output our nice JSON object, pretty printed please */
+				serialized_string = json_serialize_to_string_pretty(js);
+				fprintf(stdout, "%s\n", serialized_string);
+
+				/* free intermediate memory */
+				json_free_serialized_string(serialized_string);
+				json_value_free(js);
```

Comment:
we are not free in other json values like jsPostgres and jsKConfig

Reply:
I have worked under the assumptions that as soon as the values are nested into each other, the `json_value_free` will walk the graph... a quick glance at the parson source code seems to agree with that.